File: cache/src/main/java/com/mm/http/ServiceMethod.java
package com.mm.http;


import java.lang.reflect.Method;

import javax.annotation.Nullable;

/**
 * Created by : majian
 * Date : 2021/8/24
 */

abstract class ServiceMethod<T> {

    static <T> ServiceMethod<T> parseAnnotations(RetrofitCache retrofit, Class<?> service, Method method) {
        RequestFactory requestFactory = RequestFactory.parseAnnotations(retrofit, service, method);
        return HttpServiceMethod.parseAnnotations(retrofit, method, requestFactory);
    }

    abstract @Nullable
    T invoke(Object[] args);
}

File: cache/src/main/java/com/mm/http/ServiceMethod.kt
package com.mm.http

import java.lang.reflect.Method

/**
 * Created by : majian
 * Date : 2021/8/24
 */
abstract class ServiceMethod<T> {

    abstract operator fun invoke(args: Array<Any>?): T?

    companion object {
        @JvmStatic
        fun <T> parseAnnotations(
            retrofit: RetrofitCache,
            service: Class<*>,
            method: Method
        ): ServiceMethod<T> {
            val requestFactory = RequestFactory.parseAnnotations(retrofit, service, method)
            return HttpServiceMethod.parseAnnotations<Any, T>(retrofit, method, requestFactory)
        }
    }
}